### ISSUE TITLE
Region Selection + Clipboard Copy

### ISSUE THREAD

**Issue by jwilm** [jwilm/alacritty#6]
Using the mouse, it should be possible to select a region of characters and copy them onto the clipboard.

**Comment by jwilm** [comment]
Landed; Cmd-C works on macOS, and the primary selection is used automatically on X11 systems

**Comment by daurnimator** [comment]
This didn't seem to work for me on linux?

**Comment by jwilm** [comment]
Do you have xclip installed? We don't have fully proper clipboard support on X11 yet.

**Comment by daurnimator** [comment]
> Do you have xclip installed? 

Yes

**Comment by jwilm** [comment]
Couple of questions:
- Does selecting an area with the mouse work?
- Is anything printed to Alacritty's controlling terminal when you release?

**Comment by daurnimator** [comment]
Ah sorry. My mistake, I have a slightly different issue. I'm so used to holding "Shift" when I select from my terminal.
So different issue: holding shift and selecting with mouse doesn't select.

**Comment by jwilm** [comment]
That probably deserves its own issue. Would you mind filing one?

**Comment by uri** [comment]
Is there a way to copy on selection on OS X (without hitting cmd + c)

**Comment by unphased** [comment]
@jwilm I'd like to second @uri 's request. This behavior is common to many emulators and also iTerm2. It has been ingrained in my muscle memory, and in my recent months using Alacritty (on macOS) I have found myself at least 30 times forgetting to Cmd+C and having to reopen `less` to reselect my string to copy it.

I realize that Terminal.app does not do this and requires you to Cmd+C, but can we at least have auto-copy selections as a config option?

**Comment by jwilm** [comment]
@unphased yes we can definitely do that. There's some logic in _input.rs_ which decides whether or not to auto-copy on selection. It just needs to be hooked up to a config flag.

**Comment by unphased** [comment]
Thanks, I’ll start to manually hack that before I run my builds and will produce a PR once I learn enough about how to add configs.

**Comment by unphased** [comment]
I'm having trouble finding code that obviously controls the copying of the buffer on select. As mentioned here (https://github.com/jwilm/alacritty/pull/974#issuecomment-354928130)

Now I did find [this line](https://github.com/jwilm/alacritty/blob/master/src/input.rs#L384) which seems to do it but i can't find the other thing that corresponds to the actual macOS clipboard. I realize that linux has two clipboards, and I just want for my purposes to hack my macOS build so that the "primary selection clipboard" yank on selection also populates macOS's "system clipboard". 

Maybe something in [this function](https://github.com/jwilm/alacritty/blob/master/src/event.rs#L58).

Thanks

**Comment by jwilm** [comment]
@unphased May I suggest holding off on making any changes to the selection code? There will be some pretty big changes coming to this soon.

**Comment by unphased** [comment]
I guess I can wait... 😄

**Comment by jwilm** [comment]
@unphased The main thing in my mind is scrollback. It changes a number of things in input/selection, and it's easier to hold off on these until after that.

Once that lands, 1 will be trivial to add, and I think 2 becomes a little easier than it is today.

**Comment by unphased** [comment]
Thanks, looking forward to it!

**Comment by jonlorusso** [comment]
Was this issue ever addressed after scrollback et al was merged?

**Comment by chrisduerr** [comment]
Both of these issues mentioned by unphased are resolved.

**Comment by mdiehm** [comment]
@chrisduerr could you give a hint to how on Mac it can now be achieved that selecting a text automatically copies it to the clipboard?

**Comment by kchibisov** [comment]
@mdiehm https://github.com/jwilm/alacritty/blob/3e6e8a03eeea780e28b6fc217ee495dee3342c48/alacritty.yml#L299
